edx_sga/tests/integration_tests.py

```diff
@@ -259,13 +257,15 @@ def test_student_view_with_upload(self, fragment, render_template):
         Test student is able to upload assignment correctly.
         """
         block = self.make_one()
-        self.personalize(
-            block, **self.make_student(block, 'fred"', sha1="foo", filename="foo.bar")
-        )
-        block.student_view()
-        context = render_template.call_args[0][1]
-        student_state = json.loads(context["student_state"])
-        self.assertEqual(student_state["uploaded"], {"filename": "foo.bar"})
+        student = self.make_student(block, "fred", sha1="foo", filename="foo.bar")
+        self.personalize(block, **student)
+        user = student["module"].student
+        student_id = anonymous_id_for_user(user,self.course_id)
+        with mock.patch.object(StaffGradedAssignmentXBlock.get_submission,"__defaults__",(student_id,)):
```

Comment:
Can you tell me more about why we needed this mocking? What made the existing tests break, were they not getting the `StaffGradedAssignmentXBlock` object now or was there some other reason?

Reply:
I was getting no submission object when the test was running. When I dig into it, I found that our SGA xblock is created with the [`staff user name`](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/tests/integration_tests.py#L127) in the `make_one` fn and our submission is being made with the [`student_id`](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/tests/integration_tests.py#L172). The test is using [`block.student_view()`](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/tests/integration_tests.py#L265) which is calling [`self.student_state()`](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/sga.py#L581) -> [`self.get_submission()`](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/sga.py#L664) ->  [`self.get_student_item_dict(student_id is None)`](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/sga.py#L669).  By default the student_id is None and hence the xblock sets the student_id to the staff_id [here](https://github.com/mitodl/edx-sga/blob/fix-integration-tests/edx_sga/sga.py#L655). Therefore I mock the `get_submission` method to use the student_id of the student, by which we have created the submission and not the staff id.

Reply:
Alright, So, wouldn't it make sense to create both the SGA block and the submission with Student ID when we are testing it with a student? We might probably achieve this by adding an extra param in `make_one` method it uses the staff username as default when the username isn't passed to the function, otherwise use the username passed?